### DISCUSSION TITLE
How to use clipboard in vscode extension enviroment?

### DISCUSSION THREAD

**Discussion by hemengke1997** [Slackadays/Clipboard#186] [Q&A]
Clipboard is a Grate project.

I'm a javascript developer, and currently use `execa` to invoke `cb`
Eveything works well in node env, but `paste` can't work in vscode extension env.

When I copied an image then paste it with following:

```ts
import { execa } from 'execa'

execa('cb', ['paste'], { stdout: 'inherit' })
```
or
```ts
import { spawn } from 'node:child_process'

spawn('cb', ['paste'], { stdout: 'inherit' })
```

nothing happends. If I change `stdout` to `pipe`, execa will output image stream.

<img width="965" alt="image" src="https://github.com/Slackadays/Clipboard/assets/49073282/8ad468f6-008f-4bfc-8732-de41e1a78059">

Is there any method to output image directly instead of stream? (It's hard to deal with clipboard case)

Thank you!

**Comment by Slackadays** [comment]
That's weird, because it should only output the raw data and not the formatted version. In fact, the only places it does so are in features that you aren't using right now. Could it be the vscode extension env that's making the output look like that?

**Reply by hemengke1997** [reply]
> Could it be the vscode extension env that's making the output look like that?

Same thinking. It's vscode behavior though, but I have no idea to solve this..
